bundles/org.openhab.core.model.core/src/main/java/org/openhab/core/model/core/internal/ModelRepositoryImpl.java

```diff
@@ -98,20 +98,22 @@ public ModelRepositoryImpl(final @Reference SafeEMF safeEmf) {
 
     @Override
     public boolean addOrRefreshModel(String name, final InputStream originalInputStream) {
+        logger.info("Loading model '{}'", name);
         Resource resource = null;
-        InputStream inputStream = null;
-        try {
-            if (originalInputStream != null) {
-                byte[] bytes = originalInputStream.readAllBytes();
-                String validationResult = validateModel(name, new ByteArrayInputStream(bytes));
-                if (validationResult != null) {
-                    logger.warn("Configuration model '{}' has errors, therefore ignoring it: {}", name,
-                            validationResult);
-                    removeModel(name);
-                    return false;
-                }
-                inputStream = new ByteArrayInputStream(bytes);
+        byte[] bytes = null;
+        try (InputStream inputStream = originalInputStream) {
```

Comment:
ftr: This is probably not necessary as the `FolderObserver` handles the passed `InputStream` (`originalInputStream`), but I do not think that it will hurt:

https://github.com/openhab/openhab-core/blob/b7ab8070789079a9a9cbd1091ea7b9152f1a04e1/bundles/org.openhab.core.model.core/src/main/java/org/openhab/core/model/core/internal/folder/FolderObserver.java#L233-L238

Reply:
True, I didn't see that. But imho it is better to explicitly make sure that the stream is closed within the method. One should not rely on the caller to do it.